### PR TITLE
[WIP] Wrapping GAMs and GLMs for competing risks

### PR DESCRIPTION
**Pull request by pkopper**
Implemented working and documented functions that can handle competing risks.  See #107

### CODE REVIEW

**Comment by fabian-s** [comment]
good job, @pkopper :+1: 

as discussed today, please:
- [ ] avoid duplicated code in gam/glm fit by putting shared code into an internal fit function
- [ ] make the returned objects more human-readable by preserving factor levels for `factor`-status columns and using them for the `print` / `summary` methods

points for discussion (elsewhere):

- `predict` only up to the observed state change/last obs. period or over entire follow-up (shoudl have same behavior throughout package)?
- (how to) allow formula lists for transition-specific effects?

File: R/cr.R [hunk 1]
```diff
@@ -194,3 +169,117 @@ modify_cr_data <- function(data, cr) {
...
+}
+
+hazard_adder_cr <- function(newdata, object, hazard_function, type, ci, se_mult, 
+                            ci_type, overwrite, time_var, ...) {
+  new_cols <- as.data.frame(matrix(0, nrow = nrow(newdata), 
+                                   ncol = length(object) + 
+                                     length(object) * ci * 2))
+  j <- 1
+  for (i in 1:length(object)) {
+    if (ci) {
```
Comment (fabian-s) [inline R/cr.R]: ich glaub das if-else hier ließe sich vermeiden wenn man einmal am anfang den index j bzw die indexmenge in die man zuweist je nach dem wert von ci generiert. das wäre dann kürzer und DRY

**Comment by adibender** [comment]
- can we remove the `.RData` and `.Rproj` files. 
- I'm currently working on rewriting the code for the formula interface (#88)

**Comment by fabian-s** [comment]
haven't read this: https://onlinelibrary.wiley.com/doi/full/10.1002/sim.8399  yet, but could be relevant for this as well?

**Comment by adibender** [comment]
@pkopper is this branch up to date?

**Comment by adibender** [comment]
@pkopper Whats the status for this PR?

**Comment by pkopper** [comment]
new design, most code replicated there:

**Pull request closed without merging**